Source/SpacePark/ParkingEngine.cs

```diff
@@ -22,56 +26,303 @@ public static SpaceShip GetSpaceShipData(string input)
+                        count++;
+                        var s = ParkingEngine.GetSpaceShipData(item);
+
+                        Console.WriteLine($"{count}.{s.Name}");
                     }
+
+                    // Saves the users selection.
+                    var shipNumber = int.Parse(Console.ReadLine());
+
+                    // Actually creates the ship object.
+                    var spaceShip = SpaceShip.CreateStarshipFromAPI(person.Starships[shipNumber - 1]);
+                    person.CurrentShip = spaceShip;
+
+                    // Adds the person and ship to the database.
+                    ParkingEngine.ParkShip(person);
                 }
-                return p;
             }
-            return null;
+
+            else if (!ParkingEngine.IsValidPerson(name))
+            {
+                Console.WriteLine();
+                Console.WriteLine("Sorry you have to have been in Star Wars to park here.");
+                Thread.Sleep(2500);
+            }
+
+            else
+            {
+                Console.WriteLine();
+                Console.WriteLine("You have already parked here.");
+                Thread.Sleep(2500);
+            }
         }
 
-        public static SpaceShip CreateStarshipFromAPI(string url)
+        public static async Task CheckOut(Person p)
         {
-            var p = new SpaceShip();
-            var response = GetSpaceShipData(url);
+            if (p.HasPaid)
+            {
+                using (var context = new SpaceParkContext())
+                {
+                    // Sets the parkingspaces' shipID back to null.
+                    context.ParkingLot.Where(x => x.SpaceShipID == p.SpaceShipID)
+                        .FirstOrDefault()
+                        .SpaceShipID = null;
 
-            p.Name = response.Name;
-            p.ShipLength = response.ShipLength;
+                    // Nulls a persons current shipID
+                    await NullSpaceShipIDInPeopleTable(p, context);
 
-            return p;
+                    //Removes the curernt person from the person table
+                    context.Remove(context.People
+                        .Where(x => x.Name == p.Name)
+                        .FirstOrDefault());
+                    
+                    // Borde inte denna och den ovan se exakt lika ut?
+                    var temp = context.SpaceShips.Where(x => x.SpaceShipID == p.SpaceShipID)
+                        .FirstOrDefault();
+                    
+                    context.Remove(temp);
+
+                     context.SaveChanges();
+                }
+                Console.WriteLine();
+                Console.WriteLine("You have been checked out!");
+                Thread.Sleep(2500);
+            }
+            else
+            {
+                Console.WriteLine();
+                Console.WriteLine("Sorry you have to pay first!");
+                Thread.Sleep(2500);
```

Comment:
why write console.writeline(); "empty"
use""  \n  ""before the text you print out